### PR TITLE
Soen 87 tests for user role update back end

### PR DESCRIPTION
**Pull request by jfv-2000**
# Description

Implements/fixes [SOEN-87](https://soen390team8.atlassian.net/browse/SOEN-87)

- Test implementation for user role update as an admin in the back-end

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Unit tests

# Screenshots

|         Original          |         Updated          |
| :-----------------------: | :----------------------: |
| ** original screenshot ** | ** updated screenshot ** |

### CODE REVIEW

**Review by KelvinTrang — APPROVED** [review]
Very nice I like this !

**Review by amqkv — APPROVED** [review]
Scrumptious work💦

File: backend/controllers/admin-controller.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ const Patient = require("../models/patient");
  */
 async function updateRole(req, res) {
     if (req.body.newRole === req.body.oldRole) {
-        res.status(200);
+        await res.status(200).send("Same role");
```
Comment (NicholasYip) [inline backend/controllers/admin-controller.js:12]: what does this change?

**Review by NicholasYip — APPROVED** [review]
good changes, nice work

File: backend/controllers/admin-controller.js (hunk 1 above)
Comment (jfv-2000) [reply backend/controllers/admin-controller.js:12]: I just put it to be safe with promises and stuff xd, shouldnt change anything except prevent any potential asynchronous problems in the future

**Pull request merged**